sdk/mobile-center-analytics/src/test/java/com/microsoft/azure/mobile/analytics/AnalyticsTest.java

```diff
@@ -395,7 +398,15 @@ public void setEnabled() {
         verify(channel, never()).enqueue(any(Log.class), eq(analytics.getGroupName()));
 
         /* Enable back, testing double calls. */
-        Analytics.setEnabled(true);
+        final CountDownLatch latch = new CountDownLatch(1);
+        Analytics.setEnabled(true).thenAccept(new MobileCenterConsumer<Void>() {
+
+            @Override
+            public void accept(Void aVoid) {
+                latch.countDown();
+            }
+        });
+        assertTrue(latch.await(0, TimeUnit.MILLISECONDS));
```

Comment:
It's just additional testing to make sure there is no deadlock or any issue by doing that as well.